conans/client/cmd/export.py

```diff
@@ -15,29 +15,25 @@
 from conans.model.scm import detect_repo_type
 from conans.paths import CONANFILE
 from conans.search.search import search_recipes, search_packages
-from conans.util.files import is_dirty, load, mkdir, rmdir, save, set_dirty, remove
+from conans.util.files import is_dirty, load, rmdir, save, set_dirty, remove
 from conans.util.log import logger
 
 
-def export_alias(reference, target_reference, cache, output):
-    if reference.name != target_reference.name:
-        raise ConanException("An alias can only be defined to a package with the same name")
+def export_alias(ref_layout, target_reference, output, revisions_enabled):
     conanfile = """
 from conans import ConanFile
 
 class AliasConanfile(ConanFile):
     alias = "%s"
-""" % target_reference.full_repr()
+""" % target_reference
 
-    export_path = cache.export(reference)
-    mkdir(export_path)
-    save(os.path.join(export_path, CONANFILE), conanfile)
-    mkdir(cache.export_sources(reference))
-    digest = FileTreeManifest.create(export_path)
-    digest.save(export_path)
+    save(ref_layout.conanfile(), conanfile)
+    digest = FileTreeManifest.create(ref_layout.export())
+    digest.save(folder=ref_layout.export())
 
     # Create the metadata for the alias
-    _update_revision_in_metadata(cache, output, None, reference, digest)
+    _update_revision_in_metadata(ref_layout=ref_layout, revisions_enabled=revisions_enabled,
```

Comment:
Changing this to use layouts should have been done in the other PR, not in this one.